### PR TITLE
 Core/Quest: implement usage of QUEST_FLAGS_RAID to allow a quest to be completed while in raid

### LINKED ISSUE #17375: [3.3.5] Errors with some weekly quest in Wintergrasp
**Description**:
Some of the Wintergrasp weekly quests can not be completed during the event because when it starts all go into a raid group and not to a bg group.
**Current behaviour**: Tell us what happens
The quest Defend the Siege, Stop the Siege, Southern Sabotage, Toppling the Towers and Victory in Wintergrasp can not be completed during the event because all players go to a raid group and not to a .bg group.
**Expected behaviour**: Tell us what should happen instead
During the WIntergrasp event all players in it  must enter a bg group not in a raid group.

**Steps to reproduce the problem**:
1. Go to Wintergraps when the event is running.
2. Try to say any in /bg
3. The game show "You not are in a Battleground group".

**Branch(es)**: 
3.3.5 
**TC hash/commit**:
TrinityCore rev. 7427ca5f8e5b+ 2016-05-24
**TDB version**:
335.61
**Operating system**:
W7 64bit

I for a temporal solution changed the QuestInfoID to 62 so that they can be completed in a raid group. Sorry my bad english.

```sql 
UPDATE `quest_template` SET `QuestInfoID`=62 WHERE `ID` IN (13181, 13183, 13185, 13186, 13222, 13223, 13538, 13539);

### PR DESCRIPTION
**Pull request by SnapperRy**
**Changes proposed**:

Some quests (namely, only the Wintergrasp weekly quests, as far as I know) aren't flagged as RAID QUEST clientside, but should still be completable in a raid.

So, a SpecialFlag seems ideal to me for this job.

**Target branch(es)**: 335

**Issues addressed**: Closes #17375, Updates #7953

**Tests performed**: tested and working.

### CODE REVIEW

File: src/server/game/Quests/QuestDef.cpp [hunk 1]
```diff
@@ -285,6 +285,9 @@ bool Quest::IsAllowedInRaid(Difficulty difficulty) const
     if (IsRaidQuest(difficulty))
         return true;
 
+    if (SpecialFlags & QUEST_SPECIAL_FLAGS_ALLOWED_IN_RAID)
```
Comment (DDuarte) [inline src/server/game/Quests/QuestDef.cpp]: `if ((SpecialFlags & QUEST_SPECIAL_FLAGS_ALLOWED_IN_RAID) != 0)`

There was a compiler that had a warning due to implicit conversion to bool, better make it explicit

Comment (SnapperRy) [inline src/server/game/Quests/QuestDef.cpp]: Done, thanks :P didn't know about that.

**Comment by ForesterDev** [comment]
These quests have flag QUEST_FLAGS_RAID. I think the better way is use this flag

`diff --git a/src/server/game/Quests/QuestDef.cpp b/src/server/game/Quests/QuestDef.cpp
index cb3a6d6..f19cf72 100644
--- a/src/server/game/Quests/QuestDef.cpp
+++ b/src/server/game/Quests/QuestDef.cpp
@@ -277,6 +277,9 @@ bool Quest::IsRaidQuest(Difficulty difficulty) const
             break;
     }
 
+    if (Flags & QUEST_FLAGS_RAID)
+        return true;
+
     return false;
 }
 
diff --git a/src/server/game/Quests/QuestDef.h b/src/server/game/Quests/QuestDef.h
index ee803a2..23a7792 100644
--- a/src/server/game/Quests/QuestDef.h
+++ b/src/server/game/Quests/QuestDef.h
@@ -134,7 +134,7 @@ enum QuestFlags
     QUEST_FLAGS_SHARABLE                = 0x00000008,   // Can be shared: Player::CanShareQuest()
     QUEST_FLAGS_HAS_CONDITION           = 0x00000010,   // Not used currently
     QUEST_FLAGS_HIDE_REWARD_POI         = 0x00000020,   // Not used currently: Unsure of content
-    QUEST_FLAGS_RAID                    = 0x00000040,   // Not used currently
+    QUEST_FLAGS_RAID                    = 0x00000040,   // Can be completed while in raid
     QUEST_FLAGS_TBC                     = 0x00000080,   // Not used currently: Available if TBC expansion enabled only
     QUEST_FLAGS_NO_MONEY_FROM_XP        = 0x00000100,   // Not used currently: Experience is not converted to gold at max level
     QUEST_FLAGS_HIDDEN_REWARDS          = 0x00000200,   // Items and money rewarded only sent in SMSG_QUESTGIVER_OFFER_REWARD (not in SMSG_QUESTGIVER_QUEST_DETAILS or in client quest log(SMSG_QUEST_QUERY_RESPONSE))
`

**Comment by ForesterDev** [comment]
All of these quests have flag QUEST_FLAGS_RAID. I think, the better way is use it

https://gist.github.com/ForesterDev/430b42e6e00c8cb617dab3eae090b452

**Comment by SnapperRy** [comment]
You're right, thanks! Didn't notice that flag before.

Updated.

**Pull request merged**